sky/provision/runpod/instance.py

```diff
@@ -224,11 +225,19 @@ def query_ports(
 ) -> Dict[int, List[common.Endpoint]]:
     """See sky/provision/__init__.py"""
     del head_ip, provider_config  # Unused.
-    instances = _filter_instances(cluster_name_on_cloud, None, head_only=True)
-    assert len(instances) == 1
-    head_inst = list(instances.values())[0]
-    return {
-        port: [common.SocketEndpoint(**endpoint)]
-        for port, endpoint in head_inst['port2endpoint'].items()
-        if port in resources_utils.port_ranges_to_set(ports)
-    }
+    # RunPod ports sometimes take a while to be ready.
+    start_time = time.time()
+    while True:
```

Comment:
Let's add a timeout, in case there is any issue with the ports. For example, the docker image does not expose ssh port.

Reply:
There is already a 30 second timeout..?

Reply:
Ahh, I missed that. Just wondering if we should raise an error or print and warning for that, instead of silently ignore?

Reply:
Good point! Added. Thanks!